UNSTATED_3_RFC.md

```diff
@@ -0,0 +1,138 @@
+# Unstated 3.0 RFC
+
+I wanted to create an RFC for Unstated 3.0 which I want to take advantage of
+the new [React Hooks](https://reactjs.org/docs/hooks-intro.html) APIs.
+
+## Example
+
+My idea for it right now looks like this:
+
+```js
+import React, { useState } from 'react';
+import { Provider, useProvided, useInject } from 'unstated';
+
+function useCounter(initialCount = 0) {
+  let [count, setCount] = useState(initialCount);
+  let reset = () => setCount(0);
+  let increment = () => setCount(count => count - amount);
+  let decrement = () => setCount(count => count + amount);
+  return { count, reset, increment, decrement };
+}
+
+function Counter() {
+  // use existing instance from context, or create new one
+  let counter = useProvided(useCounter); 
```

Comment:
for some reason `useProvided` doesn't looks like a good name, may be like `useGlobalState` / `useStateFrom`/ `useInstance`

Reply:
How can we create an instance of a hook?